### PR TITLE
Move from `pdm-pep517` to `pdm-backend` due to the rename.

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

**Comment by KOLANICH** [comment]
@fjarri

File: pyproject.toml [hunk 1]
```diff
@@ -49,8 +49,8 @@ lint = [
 ]
 
 [build-system]
-requires = ["pdm-pep517"]
```
Comment (fjarri) [inline pyproject.toml:52]: Could you point me to the info about the rename? The docs say that it is still "pdm-pep517" for the current version: https://pdm.fming.dev/latest/pyproject/build/ . `pdm build` works fine too with `pdm` 2.4.7.

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [project]
 name = "peval"
 version = "0.2.0.dev"
-license = "MIT"
+license = {text = "MIT"}
```
Comment (fjarri) [inline pyproject.toml:4]: What problems does it cause? I have this set in my other projects, and they published just fine only a few weeks ago.

**Comment by fjarri** [comment]
Sorry, I was occupied with work and forgot about this PR.

**Comment by KOLANICH** [comment]
1. on PyPI https://pypi.org/project/pdm-pep517/ repository link is set to https://github.com/pdm-project/pdm-pep517 , which redirects to https://github.com/pdm-project/pdm-backend , in which ReadMe it is said

>NOTE This project has been renamed from pdm-pep517 and the old project lives in the legacy branch.

2. In PEP 621 it is said

>license
>Format: Table

>The table may have one of two keys. The `file` key has a string value that is a relative file path to the file which contains the license for the project. Tools MUST assume the file’s encoding is UTF-8. The `text` key has a string value which is the license of the project whose meaning is that of the License field from the core metadata. These keys are mutually exclusive, so a tool MUST raise an error if the metadata specifies both keys.

So `pdm-backend` follows PEP 621 more strictly, the same way other backends, like `setuptools` do.

**Comment by KOLANICH** [comment]
>`pdm build` works fine too with pdm 2.4.7.

`pdm build` is not how building wheels should be done. `pdm build` can call not PEP 517 backend, but `pdm` internal implementation different from the one implemented in `pdm-backend`.

**Comment by fjarri** [comment]
> NOTE This project has been renamed from pdm-pep517 and the old project lives in the legacy branch.

Fair enough. I would wait until it officially appears in the PDM docs, but I guess it is just a matter of time. The build still works, at lest.

> The table may have one of two keys. 

The thing is, the legacy format seems to work fine with PyPI (e.g. https://pypi.org/project/grunnur/ was published just a month ago). On the other hand, I remember switching to the new format in another project and getting errors during upload to PyPI, so I had to roll it back. But that was a year ago, so maybe they have fixed the issue. 

> `pdm build` is not how building wheels should be done.

Why not?

**Comment by KOLANICH** [comment]
>legacy format seems to work fine with PyPI

PyPI doesn't use `pyproject.toml`, it uses standardised metadata kinda the one in a wheel. `*.tar.gz` archives on PyPI are not arbitrary archives, they contain a dir with metadata.

**Comment by KOLANICH** [comment]
>Why not?

Because behavior of "fat tools" (`pdm`, `flit`, `hatch`, `poetry`( can be different from the one of their corresponding thin backends (`pdm-backend`, `flit-core`, `hatchling`, `poetry-core`) used for building packages by `build` and `pip`. TBH `hatch` uses `hatchling` under the hood, but I see no reason why the tools cannot just rely on the backends specified in `pyproject.toml` instead, it would allow using any backend, for example `setuptools` while still using the "fat tool" ot choice. I guess it is just a matter of promotion of backends associated with the tools, a NiH-syndrome.

**Comment by fjarri** [comment]
I don't think it's that important since I'm the only person doing the builds anyway. But `python -m build` produces the same results in any case. Relying on `setuptools` has its appeal, but the fact that it seemingly releases a new major version every week makes me nervous.

**Comment by KOLANICH** [comment]
>I don't think it's that important since I'm the only person doing the builds anyway. 

It's not true, I usual,y install libs from their sources, not PyPI. In fact I use this lib within my package manager intended to be a replacement for `pip` and PyPI.

>Relying on setuptools has its appeal, but the fact that it seemingly releases a new major version every week makes me nervous.

It is cool that `setuptools` devslopment has not stalled. I hope that somewhen we can get rid of `setup.py` completely and prohibit it for the most of packages and create a whitelist of build backends, this way making building of packages secure, in the sense that malicious tricks within it cannot be possible.

**Pull request merged**